src/IECoreGL/Selector.cpp

```diff
@@ -342,13 +359,24 @@ class Selector::Implementation : public IECore::RefCounted
 			glViewport( m_prevViewport[0], m_prevViewport[1], m_prevViewport[2], m_prevViewport[3] );
 			m_frameBufferBinding.reset();
 
-			IECoreImage::ImagePrimitivePtr idsImage = m_frameBuffer->getColor()->imagePrimitive();
+			IECoreImage::ImagePrimitivePtr idsImage = m_frameBuffer->getColor( g_idBufferIndex )->imagePrimitive();
 			const IECore::UIntVectorData *idsData = static_cast<const IECore::UIntVectorData *>( idsImage->channels["Y"].get() );
 			const std::vector<unsigned int> ids = idsData->readable();
 
-			IECoreImage::ImagePrimitivePtr zImage = m_frameBuffer->getDepth()->imagePrimitive();
-			const IECore::FloatVectorData *zData = static_cast<const IECore::FloatVectorData *>( zImage->channels["Z"].get() );
-			const std::vector<float> z = zData->readable();
+			std::vector<float> depth;
+
+			if( !m_useCameraDepth )
+			{
+				IECoreImage::ImagePrimitivePtr zImage = m_frameBuffer->getDepth()->imagePrimitive();
+				const IECore::FloatVectorData *zData = static_cast<const IECore::FloatVectorData *>( zImage->channels["Z"].get() );
+				depth = zData->readable();
```

Comment:
I see that we were doing this before, but we shouldn't really need to take a copy here. We should be able to keep the FloatVectorDataPtr alive and then just reference its data.

Reply:
Fixed in https://github.com/ImageEngine/cortex/pull/1404/commits/2e755a3c660a3ebcc112defc29d45f945144db4c. Should be a clearer diff when it's squashed down.